packages/babylon/src/plugins/typescript.js

```diff
@@ -802,10 +797,11 @@ export default (superClass: Class<Parser>): Class<Parser> =>
     tsTryParseTypeArgumentsInExpression(): ?N.TsTypeParameterInstantiation {
       return this.tsTryParseAndCatch(() => {
         const res: N.TsTypeParameterInstantiation = this.startNode();
-        this.expectRelational("<");
-        const typeArguments = this.tsParseDelimitedList(
-          "TypeParametersOrArguments",
-          this.tsParseType.bind(this),
+        const typeArguments = this.tsExpectLessThanThenParseInType(() =>
+          this.tsParseDelimitedList(
```

Comment:
I think that this would be more redable, since long names can be hard to read:
```js
const typeArguments = this.tsInType(() => {
  this.expectRelational("<");
  this.tsParseDelimitedList(
```